lib/hnvm/config.sh

```diff
@@ -12,7 +12,42 @@ while [ -h "$source" ]; do # resolve $source until the file is no longer a symli
 done
 script_dir="$( cd -P "$( dirname "$source" )" >/dev/null 2>&1 && pwd )"
 
-export COMMAND_OUTPUT=${HNVM_OUTPUT_DESTINATION:-/dev/stdout}
+COMMAND_OUTPUT=""
+
+# Try aggressively to find an environment-supported, redirectable, stdout fd descriptor
+#
+# In nodejs spawned subprocess (think npm pre/postinstall scripts), stdio can be
+# provided as sockets, which is not able to do simple '>>' bash redirection
+# https://nodejs.org/api/child_process.html#optionsstdio
+# "These are not actual Unix pipes and therefore the child process can not use them by their
+# descriptor files, e.g. /dev/fd/2 or /dev/stdout."
```

Comment:
Relevant art
- https://stackoverflow.com/questions/72858721/error-no-such-device-or-address-from-child-process-exec-pipe-helps-why
- https://stackoverflow.com/questions/37618443/why-cant-bash-recognize-the-existence-of-a-socket-file
- Especially this explanation: https://stackoverflow.com/a/72906798
- nodejs docs https://nodejs.org/api/child_process.html#optionsstdio

I reworked things to try to entirely avoid using sockets as targets of our bash redirects. I also added tests to make sure hnvm still works (by sending to `/dev/null`), in the rare case that the only other options are sockets.

---
Originally noticed in https://compass-tech.atlassian.net/browse/TECH-7725

I noticed similar crash behavior to the ticket, [in CI test runs for this pr](https://app.circleci.com/pipelines/github/UrbanCompass/hnvm/104/workflows/830b67dd-11bb-46ab-82c3-023130e636b5/jobs/149?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary), so I added the new test to make sure the problem is resolved in our CI runs